File: demo2/src/ContosoCrafts.WebSite/Startup.cs
using System;
using ContosoCrafts.WebSite.Services;
using EventAggregator.Blazor;
using Microsoft.AspNetCore.Builder;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;

namespace ContosoCrafts.WebSite
{
    public class Startup
    {
        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }

        public IConfiguration Configuration { get; }

        public void ConfigureServices(IServiceCollection services)
        {
            services.AddRazorPages();
            services.AddServerSideBlazor();
            services.AddHttpClient("dapr", c =>
            {
                c.BaseAddress = new Uri("http://localhost:3500");
                c.DefaultRequestHeaders.Add("User-Agent", typeof(Program).Assembly.GetName().Name);
            });
            services.AddControllers();
            services.AddScoped<IEventAggregator, EventAggregator.Blazor.EventAggregator>();
            services.AddSingleton<IProductService, DaprProductService>();
        }

        public void Configure(IApplicationBuilder app)
        {
            app.UseExceptionHandler("/Error");

            app.UseHsts();
            app.UseStaticFiles();

            app.UseRouting();
            app.UseEndpoints(endpoints =>
            {
                endpoints.MapRazorPages();
                endpoints.MapControllers();
                endpoints.MapBlazorHub();
            });
        }
    }
}

File: demo2/src/ContosoCrafts.CheckoutProcessor/Startup.fs
namespace ContosoCrafts.CheckoutProcessor

// left semicolons to be more C#-ish
open Microsoft.AspNetCore.Builder;
open Microsoft.Extensions.Configuration;
open Microsoft.Extensions.DependencyInjection;

type Startup(configuration: IConfiguration) =

    member this.Configuration = configuration;

    member this.ConfigureServices(services: IServiceCollection) =
        services.AddControllers();

    member this.Configure(app: IApplicationBuilder) =
        // warning here for not using a proper ignore
        app.UseRouting();
        app.UseEndpoints(fun endpoints ->
            // no ignore here would be an error for a lambda function
            endpoints.MapControllers()
            |> ignore<ControllerActionEndpointConventionBuilder>
        );